lib/util/src/bonsai_tree.c

```diff
@@ -453,6 +452,7 @@ bn_find_impl(struct bonsai_root *tree, const struct bonsai_skey *skey, enum bons
     bounds = atomic_read_acq(&tree->br_bounds);
     if (bounds > 0) {
         struct bonsai_kv *bkv = rcu_dereference(tree->br_kv.bkv_prev); /* max key */
+        uint lcp;
```

Comment:
I really hate how we use uint all over the place for this stuff when we mark the API as size_t. What about LTO made this not work? Not obvious to me.

Reply:
key_imm_kley() returns u32, which i'd rather it return uint (it'll never be longer than a handful of bytes, so uint is the natural word size of most machines these days, and we wouldn't compile nor run on any machine where uint was smaller than 4-bytes).
I don't know why lto build triggers the warning about memcmp() out of bounds, but it did, and it was correct.  Why the warning isn't always triggered is disturbing..

Reply:
Was the error triggered during link time or was this caught at runtime? Wondering if we should be building and testing LTO?

Reply:
During compile time, it was just a warning that didn't cause the build to fail (IIRC).
We prob should build and test LTO, but I opened another ticket because if you build LTO then hse kvdb create fails...
Yet, it worked recently, I used to build and run it...